src/AppBundle/Entity/LicenseRepository.php

```diff
@@ -134,6 +134,91 @@ public function findRecent()
         return $result;
     }
 
+    private function Erf($x) {
+        $pi = 3.141592;
+        $a = 0.140012;
+        return ($x > 0) - ($x < 0) * sqrt(1-exp(-$x**2*(4/$pi+$a*$x**2)/(1+$a*$x**2)));
+    }
+
+    private function NormalCdf($x) {
+        $loc = -12.0;
+        $scale = 31.0;
+        return 0.5*(1+$this->Erf(($x-$loc)/($scale*sqrt(2))));
+    }
+
+    private function getExpectedValue($endDate, $renewalType) {
+        $yesterday = new \DateTime('-1 day');
+        $endofmonth = new \DateTime('last day of this month');
+
+        if ($renewalType == "AUTO_RENEW") {
+            if (($endDate >= $yesterday) and ($endDate <= $endofmonth)) {
+                return 1.0;
+            } else {
+                return 0.0;
+            }
+        } else {
+            // Probability for renewal of a license today given the end date relative to today (normal distribution)
+            $renewalRate = 0.9;
+            $endOfMonthOffset = (int)$endDate->diff($endofmonth)->format("%r%a");
+
+            return $renewalRate * $this->NormalCdf($endOfMonthOffset) / 30 * ((new \DateTime())->diff($endofmonth)->days+1);
+        }
+    }
+
+    private function getPrice($addonKey, $userCnt, $renewalLength) {
+        return $this->getEntityManager()->getRepository('AppBundle:Price')->getRenewalPrice($addonKey, $userCnt, $renewalLength);
+    }
+
+    public function findEstimatedMonthlyIncome()
+    {
+        $licenses = $this->createQueryBuilder('l')
+            ->where('l.licenseType in (?1)')
+            ->andWhere('l.endDate >= :beginning')
+            ->andWhere('l.endDate <= :end')
+            ->setParameter('1', array_values(['COMMERCIAL', 'STARTER', 'ACADEMIC']))
+            ->setParameter('beginning', new \DateTime('-150 days'))
+            ->setParameter('end', new \DateTime('+175 days'))
+            ->getQuery()
+            ->getResult();
+
+        $yesterday = new \DateTime('-1 days');
+        $endofmonth = new \DateTime('last day of this month');
+
+        $total = 0;
+        foreach ($licenses as $license) {
+            $lastSales = $this->getEntityManager()->getRepository('AppBundle:Sale')->findLastSalesByLicenses(array($license));
+            $lastSale = reset($lastSales);
```

Comment:
When `$lastSales` is empty array, `reset()` returns false.

The login then fails later